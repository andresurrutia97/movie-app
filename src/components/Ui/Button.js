import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, onClick, type }) => {
  const baseClass = 'w-full focus:outline-none text-white py-2 px-4 rounded-md';

  let classes;

  switch (type) {
    case 'outlined':
      classes = [baseClass, 'border border-2 custom-blue-border bg-black'].join(
        ' '
      );
      break;

    default:
      classes = [baseClass, 'custom-blue-bg'].join(' ');
  }

  return (
    <button type="button" onClick={onClick} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};

Button.defaultProps = {
  children: 'Click me',
  onClick: () => {},
  type: '',
};

export default Button;
