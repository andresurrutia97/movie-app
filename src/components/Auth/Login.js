import React from 'react';
import PropTypes from 'prop-types';
import { withRouter, Link } from 'react-router-dom';

import Button from '../Ui/Button';
import logo from '../../img/logoWeb.png';

const Login = ({ login }) => (
  <div className="w-full flex flex-col text-white items-center px-6 ">
    <img className="w-56 mb-2" src={logo} alt="logo" />
    <span className="mb-2 text-white text-center text-4xl font-semibold md:text-5xl xl:text-6xl">
      The best streaming platform
    </span>
    <span className="text-white text-center md:text-xl">
      Enjoy wherever you are.
    </span>
    <div className="w-full md:w-64 mt-12">
      <Button onClick={() => login()}>LOGIN</Button>
    </div>
    <Link className="text-sm mt-4" to="/toc">
      TOC & Privacy policy
    </Link>
  </div>
);

Login.propTypes = {
  login: PropTypes.func,
};

Login.defaultProps = {
  login: () => {},
};

export default withRouter(Login);
