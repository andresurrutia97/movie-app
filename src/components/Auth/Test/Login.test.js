import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Login from '../Login';
import Button from '../../Ui/Button';

describe('<Login/>', () => {
  const loginFn = jest.fn();
  const component = (
    <BrowserRouter>
      <Login login={loginFn} />
    </BrowserRouter>
  );

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call the login function when click on login button', () => {
    const wrapper = mount(component);
    wrapper.find(Button).simulate('click');
    expect(loginFn).toBeCalledTimes(1);
  });
});
