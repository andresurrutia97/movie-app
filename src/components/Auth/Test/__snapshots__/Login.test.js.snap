// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render correctly 1`] = `
<div
  className="w-full flex flex-col text-white items-center px-6 "
>
  <img
    alt="logo"
    className="w-56 mb-2"
    src="logoWeb.png"
  />
  <span
    className="mb-2 text-white text-center text-4xl font-semibold md:text-5xl xl:text-6xl"
  >
    The best streaming platform
  </span>
  <span
    className="text-white text-center md:text-xl"
  >
    Enjoy wherever you are.
  </span>
  <div
    className="w-full md:w-64 mt-12"
  >
    <button
      className="w-full focus:outline-none text-white py-2 px-4 rounded-md custom-blue-bg"
      onClick={[Function]}
      type="button"
    >
      LOGIN
    </button>
  </div>
  <a
    className="text-sm mt-4"
    href="/toc"
    onClick={[Function]}
  >
    TOC & Privacy policy
  </a>
</div>
`;
